services/lgtm/lgtm-base.js

```diff
@@ -0,0 +1,29 @@
+'use strict'
+
+const Joi = require('joi')
+const BaseJsonService = require('../base-json')
+
+const schema = Joi.object({
+  alerts: Joi.number(),
+
+  languages: Joi.array(),
```

Comment:
Can we make both of these required. Also, if we've got an array of objects, we should validate the contents of that array, so if we're expecting the elements to have keys `lang` and `grade`, that should be validated in the schema

Reply:
Thanks so much for updating this!

I think this is the only thing still outstanding. You can use something like this should work:

```js
Joi.array()
  .items(Joi.object({
    lang: Joi.string().requred(),
    grade: Joi.equal('A', 'B', 'C', ...).required()
  }))
  .required()
```

I think `.items().required()` implies `.min(1)`, oddly enough.

Reply:
We could do that, but Joi will only return the first language as each language is a separate object, so we need a way for Joi to validate more than one object without knowing how many there will be.

Reply:
if its wrapped in `.array().items()`, Joi will pass multiple objects:

```
$ node

> const Joi = require('joi')
undefined

> const schema = Joi.array().items(Joi.object({
...   lang: Joi.string().required(),
...   grade: Joi.string().required()
... })).required()
undefined

> const data = [
...   { lang: 'javascript', grade: 'A' },
...   { lang: 'python', grade: 'A' },
...   { lang: 'php', grade: 'A' },
... ]
undefined

> Joi.validate(data, schema, { stripUnknown: true })
{ error: null,
  value: 
   [ { lang: 'javascript', grade: 'A' },
     { lang: 'python', grade: 'A' },
     { lang: 'php', grade: 'A' } ],
  then: [Function: then],
  catch: [Function: catch] }
```

Reply:
You could also experiment here: https://runkit.com/melnikow/joi-schema-testing

Reply:
Ah, gotcha, yea, no need to make `grade` required.